### PR TITLE
Rubocop upgrade

### PR DESCRIPTION
**Pull request by bennacer860**
upgrade rubocop to 0.32.1 to run on rails 2.2.  all the details are explained in this [issue](https://github.com/rspec/rspec-dev/issues/134)

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -8,7 +8,7 @@ branch = File.read(File.expand_path("../maintenance-branch", __FILE__)).chomp
   if File.exist?(library_path) && !ENV['USE_GIT_REPOS']
     gem lib, :path => library_path
   else
-    gem lib, :git => "git://github.com/rspec/#{lib}.git", :branch => branch
+    gem lib, :git => "https://github.com/rspec/#{lib}.git", :branch => branch
```
Comment (cupakromer) [inline Gemfile]: Could you revert this in the pull?

File: lib/rspec/core/configuration.rb [hunk 2]
```diff
@@ -1712,7 +1712,8 @@ def extract_location(path)
 
         if match
           captures = match.captures
-          path, lines = captures[0], captures[1][1..-1].split(":").map { |n| n.to_i }
+          path = captures[0]
+          lines =  captures[1][1..-1].split(":").map(&:id)
```
Comment (cupakromer) [inline lib/rspec/core/configuration.rb]: Looks like two issues here:

- double whitespace between `=` and `captures` in `lines =  captures[1]`
- the 2nd captures changed the block it needs to use `&:to_i` such as the following:

  ```ruby
  captures[1][1..-1].split(":").map(&:to_i)
  ```

File: lib/rspec/core/formatters.rb [hunk 3]
```diff
@@ -168,9 +168,9 @@ def add(formatter_to_use, *paths)
 
     def find_formatter(formatter_to_use)
       built_in_formatter(formatter_to_use) ||
-      custom_formatter(formatter_to_use)   ||
-      (raise ArgumentError, "Formatter '#{formatter_to_use}' unknown - " \
-                            "maybe you meant 'documentation' or 'progress'?.")
```
Comment (cupakromer) [inline lib/rspec/core/formatters.rb]: Did this whitespace cause the rubocop to fail? Its a little easier to read when these are aligned with the quotes.

File: lib/rspec/core/formatters/profile_formatter.rb [hunk 4]
```diff
@@ -37,7 +37,7 @@ def dump_profile_slowest_examples(profile)
           profile.slowest_examples.each do |example|
             @output.puts "  #{example.full_description}"
             @output.puts "    #{bold(Helpers.format_seconds(example.execution_result.run_time))} " \
-                         "#{bold("seconds")} #{format_caller(example.location)}"
+                         "#{bold('seconds')} #{format_caller(example.location)}"
           end
```
Comment (cupakromer) [inline lib/rspec/core/formatters/profile_formatter.rb]: We're fine with double quotes here. Just for future reference.

File: .rubocop.yml [hunk 5]
```diff
@@ -23,9 +38,20 @@ Lint/HandleExceptions:
...
+Style/SpaceAroundOperators:
+  MultiSpaceAllowedForOperators:
+      - '='
+      - '||='
+      - '==='
+      - '=>'
+      - '<<'
+
+Style/AccessModifierIndentation:
+  Enabled: false
```
Comment (cupakromer) [inline .rubocop.yml]: Can we remove this we want the modifiers using the base dedent setting.

Comment (bennacer860) [inline .rubocop.yml]: if we remove that we get something like this
```
lib/rspec/core/formatters/console_codes.rb:23:9: C: Outdent access modifiers like module_function.
        module_function
        ^^^^^^^^^^^^^^^
```
we would have to outdent `module_function`, are you ok with it?

Comment (cupakromer) [inline .rubocop.yml]: Yes, it should always have been but there was a :bug: I'm the previous rubocop

File: benchmarks/lambda_vs_nested_methods.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  x.report("nested methods") do
+    nested_methods
+  end
+
+  x.report("lambda") do
+    lambda_method
+  end
+
+  x.compare!
+end
```
Comment (myronmarston) [inline benchmarks/lambda_vs_nested_methods.rb]: Can you add the benchmark results you get?  Ideally after a `__END__` line so they can just be pasted as comments at the end of the file.  That way we can see some results w/o having to run it ourselves :).

File: benchmarks/lambda_vs_nested_methods.rb [hunk 7]
```diff
@@ -0,0 +1,23 @@
+require 'benchmark/ips'
+
+def nested_methods
+  def some_method
+  end
+end
+
+def lambda_method
+  some_lambda = lambda {|i|}
+end
```
Comment (myronmarston) [inline benchmarks/lambda_vs_nested_methods.rb]: These are not equivalent.  The lambda here is a local variable that is only visible in the method.  The nested method is adding the `some_method` method to the object when `nested_methods` is called, which mutates the object and makes the method available outside the scope of `nested_method`.  It also blows away the method cache and is generally not a good technique.

File: benchmarks/lambda_vs_nested_methods.rb (hunk 6 above)
Comment (JonRowe) [inline benchmarks/lambda_vs_nested_methods.rb]: (they're below now)

**Comment by JonRowe** [comment]
I'm not a fan of the changes to indentation here, any way we can change the rules instead?

**Comment by bennacer860** [comment]
@JonRowe are you talking about this [commit?](https://github.com/bennacer860/rspec-core/commit/0d135ddc496f3d19235d2730bd8b73efd8e7b2c9)

**Comment by cupakromer** [comment]
If he is, I believe we've agreed to keep that style (it does looks odd when it's the only access modified :grimacing:). See https://github.com/rspec/rspec-support/pull/233#discussion_r36049901 for parallel convo.

**Comment by bennacer860** [comment]
i am not sure about my next commit? should i revert back to the old indentation where access modifiers will be at the same level as a method definition.?

**Comment by cupakromer** [comment]
No change necessary. Leave them outdented (in the same column as the module name) as you did [here](https://github.com/rspec/rspec-core/pull/2051/files#diff-a78877c21b7ac7f825ac7615eb655c82L23).

**Comment by JonRowe** [comment]
I don't, although I'd rather they weren't changed I'm awaiting the others input on https://github.com/rspec/rspec-support/pull/233#discussion_r36049901 currently;

I mean the changing to spacing before operators, I know the cop is new/changed and enforcing this, but 37c6435c3db9376d1ef0ff902515aa48635079e2 changes `lib/rspec/core/formatters.rb` `lib/rspec/core/metadata.rb#L154` `lib/rspec/core/notifications.rb` away from our previous convention

**Comment by cupakromer** [comment]
> but 37c6435 changes lib/rspec/core/formatters.rb lib/rspec/core/metadata.rb#L154 lib/rspec/core/notifications.rb away from our previous convention

What was the previous convention? I wasn't aware we had one.

**Comment by JonRowe** [comment]
I always thought we choose not to intend multiline statements, and to allow aligning of operators for readability

**Comment by cupakromer** [comment]
Not indenting multiline statements sounds new to me. I've never followed that. We have allowed aligning ops for readability, though it doesn't seem to occur consistently. I believe the operator whitespace is configurable (see https://github.com/rspec/rspec-core/pull/2051/files#diff-11a0d7906801d9dea0eccb85667ad811R51) so we can surely special case more ops by default.

**Comment by myronmarston** [comment]
Consistent style has value but I'm starting to think rubocop defaults are far too stringent.   I think the value it provides may not exceed the effort we're putting in to discussing this.  I personally don't really care about most of these things and don't have the time to look into it now.  We can always disable some cops for now.  Too many cops may make it overly hard for contributors to get a green PR, anyway.

**Comment by JonRowe** [comment]
I think we should just disable it rather than add new changes to fit it at this point

**Comment by bennacer860** [comment]
Any update about the PR? is it ready to be merged or does it need more work? @cupakromer

**Comment by soulcutter** [comment]
I think this is ready to merge, although I feel like we should remove the `lambda_vs_nested_methods.rb` benchmark since it's an apples to oranges comparison. I think it was a mistaken assumption that lambdas could replace the code given a rubocop exception in 8d1088b and we probably don't need to keep that history.

Would anybody object to a rebase to remove 13b482a and ecc4bf5 (the commits for lambda_vs_nested_methods.rb)?

Also fc3560a (revert commit) seems like it should be a `fixup` into a30c5a2 .

While we're at it 056291c (fix typo) could be a `fixup` into 37c6435c3db9376d1ef0ff902515aa48635079e2

I'm happy to do this rebasing if you want.

**Comment by JonRowe** [comment]
Superseded by #2081

**Comment by JonRowe** [comment]
Thanks for kicking this off, I hope you don't mind us taking it across the line.

**Pull request closed without merging**